src/infrastructure/storage/file-manager.ts

```diff
@@ -416,6 +416,50 @@ export class FileManager implements FileManagerInterface {
     }
   }
 
+  /**
+   * Gets the maximum date available in the block numbers store.
+   * @returns The maximum date as a Date object, or null if no dates exist
+   */
+  getMaxDateFromBlockStore(): Date | null {
+    const blockNumbersData = this.readBlockNumbers();
```

Comment:
there's a bunch of duplicate code in this and getMinDateFromBlockStore. please deduplicate. Also can we rename to getMax/MinDate